pkg/image/cdi/validate.go

```diff
@@ -3,21 +3,56 @@ package cdi
 import (
 	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
 	"github.com/harvester/harvester/pkg/image/backend"
+	"github.com/harvester/harvester/pkg/image/common"
 	"github.com/harvester/harvester/pkg/webhook/types"
 )
 
 type Validator struct {
+	vmiv common.VMIValidator
 }
 
-func GetValidator() backend.Validator {
-	return &Validator{}
+func GetValidator(vmiv common.VMIValidator) backend.Validator {
+	return &Validator{vmiv}
 }
 
-func (cv *Validator) Create(_ *types.Request, _ *harvesterv1.VirtualMachineImage) error {
+func (cv *Validator) Create(req *types.Request, vmImg *harvesterv1.VirtualMachineImage) error {
+	if err := cv.vmiv.CheckDisplayName(vmImg); err != nil {
```

Comment:
Just want to confirm the current implementation of `CheckDisplayName()` at: https://github.com/harvester/harvester/blob/ce188325a591813e047d3bd5991839f704eda1f5/pkg/image/common/validator.go#L90-L95

Shouldn't it be:

```go
	for _, image := range sameDisplayNameImages {
		if vmi.Name == image.Name {
                    return werror.NewConflict("A resource with the same display name exists")
		}
                continue
	}
```

Reply:
Thanks, I do some enhancement for this checking function